Make Deployments use a ServiceAccount (#52)
m88i/nexus-operator: README.md

```

If you're running on Kubernetes, edit the Nexus resource to add a [valid host for the Ingress](#network-on-kubernetes-114) to work.

### Openshift 3.x

If you're running the Operator on Openshift 3.x it's also necessary to configure a [Security Context Constraints](https://docs.openshift.com/container-platform/3.11/admin_guide/manage_scc.html) (SCC) resource.

> This is necessary because the Nexus image the operator uses requires its container to be ran as UID 200. The use of default SCC resources in Openshift 3.x results in a failure when starting the pods, as seen in [Issue #41](https://github.com/m88i/nexus-operator/issues/41).

Reviewer: ```suggestion
This is necessary because the Nexus image requires its container to be ran as UID 200. The use of default SCC resources in Openshift 3.x results in a failure when starting the pods, as seen in [Issue #41](https://github.com/m88i/nexus-operator/issues/41).
```

> Valid SCC resources can be found at the `examples/` directory. You must associate the the SCC with the ServiceAccount in use. In the commands below it is assumed you'll be using the default ServiceAccount created when installing this Operator.

Reviewer: ```suggestion
Valid SCC resources can be found at the `examples/` directory. You must associate the SCC with the `ServiceAccount` in use. In the commands below it is assumed you'll be using the default `ServiceAccount` created when deploying a new Nexus CR.
```

For persistent configurations:

```
$ oc apply -f examples/scc-persistent.yaml
```

For volatile configurations:

```
$ oc apply -f examples/scc-persistent.yaml
```

> **Note**: you must choose one or the other, applying both will result in using the one applied last.

Once the SCC has been created, run:

```
$ oc adm policy add-scc-to-user nexus-operator -z <ServiceAccountName>
```

> This command will bind the SCC we just created with the ServiceAccount being used to create the Pods.

Reviewer: ```suggestion
This command will bind the SCC we just created with the `ServiceAccount` being used to create the Pods.
```

> If you're [using a custom ServiceAccount](##ServiceAccount), replace "`<ServiceAccountName>`" with the name of that account. If you're not using a custom ServiceAccount, the operator has created a default one which has the same name as your Nexus CR, replace "`<ServiceAccountName>`" with that.

Reviewer: ```suggestion
If you're [using a custom ServiceAccount](##Service Account), replace "`<ServiceAccountName>`" with the name of that account. If you're not using a custom ServiceAccount, the operator has created a default one which has the same name as your Nexus CR, replace "`<ServiceAccountName>`" with that.
```

drwxr-xr-x  2  200  200 4096 Apr 26 15:42 pv0001
```

> ## ServiceAccount

Reviewer: ```suggestion
## Service Account
```

> It is possible to use a custom [ServiceAccount](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/) to perform your Deployments with the Nexus Operator via:

Reviewer: ```suggestion
It is possible to use a custom [`ServiceAccount`](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/) to perform your Deployments with the Nexus Operator via:
```

>   - `spec.serviceAccountName` (*string*): ServiceAccountName is the name of the ServiceAccount to use to run the Pods. If left blank, a default ServiceAccount is created with the same name as the Nexus CR.

Reviewer: ```suggestion
  - `spec.serviceAccountName` (*string*): `ServiceAccountName` is the name of the `ServiceAccount` to run with the Pods. If left blank, a default `ServiceAccount` is created with the same name as the Nexus CR.
```

> **Important**: the Operator handles the creation of default resources necessary to run. If you choose to use a custom ServiceAccount be sure to also configure [Role](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-and-clusterrole) and [RoleBinding](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding) resources.

Reviewer: ```suggestion
**Important**: the Operator handles the creation of default resources necessary to run. If you choose to use a custom `ServiceAccount` be sure to also configure [`Role`](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-and-clusterrole) and [`RoleBinding`](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding) resources.
```
